common/.clang-format

```diff
@@ -0,0 +1 @@
+DisableFormat: true
```

Comment:
Most of `common` is our own and *should* be formatted, just `common/include/scipp/common/span` is not.

Reply:
The span folder has it's own `.clang-format` file, so I've just disabled formatting for it there.